src/pipx/commands/pin.py

```diff
@@ -0,0 +1,97 @@
+import logging
+from pathlib import Path
+from typing import Sequence
+
+from pipx.colors import bold
+from pipx.constants import ExitCode
+from pipx.emojis import sleep
+from pipx.util import PipxError
+from pipx.venv import Venv
+
+logger = logging.getLogger(__name__)
+
+
+def _update_pin_info(venv: Venv, package_name: str, is_main_package: bool, unpin: bool) -> int:
+    package_metadata = venv.package_metadata[package_name]
+    venv.update_package_metadata(
+        package_name=str(package_metadata.package),
+        package_or_url=str(package_metadata.package_or_url),
+        pip_args=package_metadata.pip_args,
+        include_dependencies=package_metadata.include_dependencies,
+        include_apps=package_metadata.include_apps,
+        is_main_package=is_main_package,
+        suffix=package_metadata.suffix,
+        pinned=not unpin,
+    )
+    return 1
+
+
+def pin(
+    venv_dir: Path,
+    verbose: bool,
+    skip: Sequence[str],
+    injected_only: bool = False,
+) -> ExitCode:
+    venv = Venv(venv_dir, verbose=verbose)
+    try:
+        main_package_metadata = venv.package_metadata[venv.main_package_name]
+    except KeyError as e:
+        raise PipxError(f"Package {venv.name} is not installed") from e
+
+    if main_package_metadata.pinned:
+        logger.warning(f"Package {main_package_metadata.package} already pinned {sleep}")
+    elif skip and not injected_only:
+        raise PipxError("--skip must be used with --injected-only")
```

Comment:
Raise or "skip skip" and log a warning?

Reply:
A warning sounds reasonable to me, since `--skip` actually implies `--injected-only`.

Reply:
>since --skip actually implies --injected-only

Why though? What if I want to upgrade the core package + all dependencies except one?

Reply:
I believe `pipx upgrade <venv> --skip <all packages except for the one you want to pin>` should work for this case.

Reply:
@dukecat0 yeah, but it doesn't seem to be convenient to list all the excluded packages... I believe it would be much simpler (and less surprising) if we don't imply `--injected-only` when using `--skip`, but allow people to use them together explicitly.

Reply:
@dukecat0 I have just tested the new `pipx` with `pin` released - and as I expected, implied `--injected-only` when using `--skip` makes it impossible to pin the main package and its dependencies except one, because if you use `--skip` - the main package won't be pinned. That's why I suggested to make these 2 options work separately.

Reply:
Sure (just random libraries to show the issue):
```
❯ pipx install black==24.3.0
❯ pipx inject black flake8 mypy
❯ pipx pin black --skip mypy
Pinned 1 packages in venv black
  - flake8 7.1.1
❯ pipx upgrade black
upgraded package black from 24.3.0 to 24.8.0 (location:
/Users/lanskij/.local/pipx/venvs/black)
```
And if I run `pipx pin black` - I would pin the main package, but also all the dependencies.

Reply:
Sounds reasonable to me, I will open a PR to change this behavior. :)

Reply:
Thanks!

Reply:
Btw, that change would also allow to introduce `--include` (as opposed to `--skip`) in the future:
`❯ pipx pin black --include flake8`

Reply:
Yes actually I was thinking about that as well, but I was not sure whether it would be redundant.

Reply:
Exactly, I found both `--skip` and `--include` flags usable, just for various use-cases